Training/Easy/ascii_art.rb

```diff
@@ -1,28 +1,41 @@
 # Solution for https://www.codingame.com/ide/puzzle/ascii-art
 class AsciiArt
-  attr_accessor :ascii_style, :letter_width, :letter_height, :word, :lines
+  attr_accessor :ascii_style, :letter_width, :letter_height, :word
 
   def initialize
-    @ascii_style = ('A'..'Z').to_a << '?'
     @letter_width = gets.to_i
     @letter_height = gets.to_i
     @word = gets.chomp
   end
 
-  def letter_index(letter)
-    index = ascii_style.find_index(letter.upcase)
-    index ? index : ascii_style.size - 1
-  end
-
   def start
-    letter_height.times do
+    Array.new(letter_height) do
       row = gets.chomp
       splitted_row = row.scan(/.{#{letter_width}}|./)
-      line = word.chars.map { |letter| splitted_row[letter_index(letter)] }.join
-      puts line
+      word.chars.map { |char| AsciiArt::Letter.new(splitted_row, char).in_ascii }.join
+    end
+  end
+
+  # Represent letter
+  class Letter
+    attr_accessor :row, :char
+
+    def initialize(row, char)
+      @row = row
+      @char = char
+    end
+
+    def letter_index(letter)
+      ascii_style = ('A'..'Z').to_a << '?'
```

Comment:
You can move it to a constant `AsciiArt::Letter::LETTERS`, so that it reserves memory once.
```ruby
LETTERS = [('A'..'Z').to_a, '?'].flatten
```